applet/main.cpp

```diff
@@ -59,13 +59,30 @@ namespace {
 
 extern "C" void userAppInit(void) {
     splInitialize();
+    spsmInitialize();
 }
 
 extern "C" void userAppExit(void) {
+    amsBpcExit();
+    spsmExit();
     splExit();
 }
 
 int main(int const argc, char const *argv[]) {
+    /* Initialize console output */
```

Comment:
The initialization block was moved here to avoid issues related to the `smExit` call required by `amsBpcInitialize`.